File: 28-mfa/python-sdk/mfa.py
import os, requests
API = "https://cost-katana-backend.store/api"
KEY = os.getenv("COST_KATANA_API_KEY")

def enable_mfa():
    res = requests.post(f"{API}/auth/mfa/enable", json={},
        headers={"Authorization": f"Bearer {KEY}"})
    res.raise_for_status()
    print(f"✅ MFA enabled. QR Code: {res.json()['data']['qrCode']}")
    return res.json()["data"]

def verify_mfa(code):
    res = requests.post(f"{API}/auth/mfa/verify",
        json={"code": code},
        headers={"Authorization": f"Bearer {KEY}"})
    res.raise_for_status()
    print("✅ MFA verified")
    return res.json()["data"]

File: 28-mfa/npm-package/mfa.ts
import axios from 'axios';
const API = 'https://api.costkatana.com/api'\;
const KEY = process.env.COST_KATANA_API_KEY;

export async function enableMFA() {
  const res = await axios.post(`${API}/auth/mfa/enable`, {},
    {headers: {'Authorization': `Bearer ${KEY}`}});
  console.log('✅ MFA enabled. QR Code:', res.data.data.qrCode);
  return res.data.data;
}

export async function verifyMFA(code: string) {
  const res = await axios.post(`${API}/auth/mfa/verify`, {code},
    {headers: {'Authorization': `Bearer ${KEY}`}});
  console.log('✅ MFA verified');
  return res.data.data;
}
